#VRML_SIM R2023b utf8

EXTERNPROTO "../protos/Field.proto"
EXTERNPROTO "../protos/Stadium_background.proto"
EXTERNPROTO "../protos/Statium_background_light.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/balls/protos/RobocupSoccerBall.proto"
EXTERNPROTO "../protos/Robot_nao.proto"

WorldInfo {
}
Viewpoint {
  orientation 0.1436442645643637 -0.09293728376312749 -0.9852557975188448 4.276583443334789
  position 2.7592985581048652 -6.265112280803773 2.5174532639514275
}
Field {
  postRadius 0.02
}
DEF RED_LEFT_ATTACKER Robot_nao {
  hidden position_0_0 -0.02220276936913699
  hidden position2_0_0 0.07918896110259128
  hidden position_0_1 0.35512422333701077
  hidden position2_0_1 -0.3498811993570804
  hidden position_0_2 0.9816005271907466
  hidden position2_0_2 0.3711174409769082
  hidden position_0_3 -2.3644942422524358e-06
  hidden position_0_4 -2.520773777155343e-06
  hidden rotation_11 0.01068939032907076 0.9628484649777798 -0.26983063285017495 0.082241090030143
  hidden rotation_18 -0.12493793537500907 0.7068729909601464 -0.6962191371653659 0.4972305370095235
  hidden position_18_0 1.031231642194313
  hidden rotation_19 0.9999999999999999 0 0 1.031231642194313
  hidden position_19_0 1.2048847927943735
  hidden rotation_20 0 0 1 1.2048847927943733
  hidden position_20_0 -0.05455770864117677
  hidden rotation_21 -1 0 0 0.05455770864117932
  hidden position_21_0 0.16611643738347995
  hidden position_21_1 0.16611643737633355
  hidden position_21_2 0.16611643693171987
  hidden rotation_22 0 1 0 0.7938835629928497
  hidden position_22_0 0.1661164381706144
  hidden rotation_23 0 0.9999999999999999 0 0.793883561093668
  hidden position_23_0 0.1661164388993073
  hidden rotation_24 0 1 0 0.7938835607635553
  hidden rotation_25 0 1 0 0.7938835629999981
  hidden position_25_0 0.16611643816598867
  hidden rotation_26 0 1 0 0.7938835610984724
  hidden position_26_0 0.16611643889823277
  hidden rotation_27 0 1 0 0.7938835607645937
  hidden rotation_28 0 1 0 0.16611643693172068
  hidden position_28_0 0.16611643619818897
  hidden rotation_29 0 -1 0 0.7938835636015298
  hidden rotation_30 0.17439569581243933 0.9290292897239504 0.3263291591585212 1.0440320340830387
  hidden position_30_0 -1.1226138378646222
  hidden rotation_31 -1 0 0 1.1226138378646222
  hidden position_31_0 -0.9704919077730838
  hidden rotation_32 0 0 -1 0.9704919077730836
  hidden position_32_0 0.38700010049555456
  hidden rotation_33 0.9999999999999999 0 0 0.387000100495555
  hidden position_33_0 0.13199999983364485
  hidden position_33_1 0.13199999980648644
  hidden position_33_2 0.1320000001454058
  hidden rotation_34 0 0.9999999999999999 0 0.8280000005426604
  hidden position_34_0 0.1320000001066408
  hidden rotation_35 0 1 0 0.827999999157757
  hidden position_35_0 0.13200000003409396
  hidden rotation_36 0 1 0 0.8279999999659058
  hidden rotation_37 0 1 0 0.8280000005697855
  hidden position_37_0 0.13200000010521093
  hidden rotation_38 0 1 0 0.8279999991593661
  hidden position_38_0 0.13200000003285123
  hidden rotation_39 0 1 0 0.8279999999671477
  hidden rotation_40 0 1 0 0.13200000014540447
  hidden position_40_0 0.13200000003223433
  hidden rotation_41 0 -1 0 0.827999999967766
  hidden rotation_42 0 -0.7071067811865475 -0.7071067811865475 2.364547134656643e-06
  hidden position_42_0 -1.5323530251869104e-06
  hidden rotation_43 -1 0 0 1.5324306645611233e-06
  hidden position_43_0 -2.7552779551067735e-06
  hidden rotation_44 0 -1 0 2.755224301790722e-06
  hidden position_44_0 -3.2968557047314837e-06
  hidden rotation_45 0 -1 0 3.29689664689274e-06
  hidden position_45_0 -3.655219598095017e-06
  hidden rotation_46 0 -1 0 3.6552521072496083e-06
  hidden position_46_0 -9.597080743974784e-07
  hidden rotation_47 -1 0 0 9.597088210645183e-07
  hidden translation_48 0.020000732187030246 -0.006999136965514907 -0.040597738768151904
  hidden rotation_48 9.905120744419884e-07 -4.232134493712115e-09 0.9999999999995096 -3.1415853138111602
  hidden rotation_52 0 -0.7071067811865476 0.7071067811865476 2.5207198207130312e-06
  hidden position_52_0 -9.544818460993402e-07
  hidden rotation_53 -1 0 0 9.546051846770714e-07
  hidden position_53_0 -2.796556144171022e-06
  hidden rotation_54 0 -0.9999999999999999 0 2.7964997739068167e-06
  hidden position_54_0 -3.5243457929143174e-06
  hidden rotation_55 0 -0.9999999999999999 0 3.5243687661068423e-06
  hidden position_55_0 -3.597579467095532e-06
  hidden rotation_56 0 -0.9999999999999999 0 3.5975736172973187e-06
  hidden position_56_0 2.63978084179032e-06
  hidden rotation_57 1 0 0 2.6398196682667916e-06
  hidden translation_58 0.020000530684696427 0.00700064002217804 -0.0405972301485041
  hidden rotation_58 9.91046910252642e-07 -2.7013116790542212e-09 0.999999999999509 -3.141585310596845
  translation -0.7565782828607793 1.8452252484655143 0.3322598888114324
  rotation -0.04488855079152709 -0.16343856586437602 -0.9855317616373571 0.0044992163455108815
  name "red_left_attacker"
  controller "red_left_attacker_controller"
}
DEF RED_RIGHT_ATTACKER Robot_nao {
  hidden position_0_0 -0.022202443297932445
  hidden position2_0_0 0.07918897193283536
  hidden position_0_1 0.35512505625915247
  hidden position2_0_1 -0.3498830457457879
  hidden position_0_2 0.9816004952222609
  hidden position2_0_2 0.37111732912365275
  hidden position_0_3 -1.3713395903755026e-06
  hidden position_0_4 -1.46092153320427e-06
  hidden rotation_11 0.010689244886191739 0.9628495058157627 -0.2698269245166325 0.08224101247463524
  hidden rotation_18 -0.12493841670585916 0.7068719067245776 -0.6962201516148417 0.4972324112713266
  hidden position_18_0 1.036476177151679
  hidden rotation_19 1 0 0 1.036476177151679
  hidden position_19_0 1.2053600814978707
  hidden rotation_20 0 0 1 1.2053600814978707
  hidden position_20_0 -0.05461458334211049
  hidden rotation_21 -0.9999999999999999 0 0 0.054614583342110175
  hidden position_21_0 0.16611783224441587
  hidden position_21_1 0.16611783223934215
  hidden position_21_2 0.16611783203796807
  hidden rotation_22 0 1 0 0.7938821681319218
  hidden position_22_0 0.16611783307841174
  hidden rotation_23 0 1 0 0.793882166185873
  hidden position_23_0 0.16611783388727497
  hidden rotation_24 0 1 0 0.7938821657755881
  hidden rotation_25 0 1 0 0.7938821681369956
  hidden position_25_0 0.16611783307410352
  hidden rotation_26 0 1 0 0.7938821661903617
  hidden position_26_0 0.1661178338861085
  hidden rotation_27 0 1 0 0.7938821657767248
  hidden rotation_28 0 1 0 0.16611783203796743
  hidden position_28_0 0.16611783123052873
  hidden rotation_29 0 -1 0 0.7938821685691905
  hidden rotation_30 0.17439564867383983 0.9290293251481194 0.3263290835007822 1.044031967882404
  hidden position_30_0 -1.1224418081731198
  hidden rotation_31 -1 0 0 1.1224418081731196
  hidden position_31_0 -0.9704304105024796
  hidden rotation_32 0 0 -0.9999999999999999 0.9704304105024798
  hidden position_32_0 0.38700010043303645
  hidden rotation_33 0.9999999999999999 0 0 0.38700010043303673
  hidden position_33_0 0.13199999984220515
  hidden position_33_1 0.13199999982165114
  hidden position_33_2 0.1320000001470008
  hidden rotation_34 0 1 0 0.828000000534112
  hidden position_34_0 0.13200000010883425
  hidden rotation_35 0 1 0 0.8279999991555635
  hidden position_35_0 0.1320000000331553
  hidden rotation_36 0 0.9999999999999999 0 0.8279999999668436
  hidden rotation_37 0 1 0 0.8280000005546289
  hidden position_37_0 0.13200000010258617
  hidden rotation_38 0 1 0 0.8279999991619917
  hidden position_38_0 0.13200000003305087
  hidden rotation_39 0 1 0 0.8279999999669474
  hidden rotation_40 0 1 0 0.13200000014700033
  hidden position_40_0 0.13200000003564008
  hidden rotation_41 0 -1 0 0.8279999999643591
  hidden rotation_42 0 -0.7071067811865476 -0.7071067811865476 1.371230729857393e-06
  hidden position_42_0 -4.2410977508931706e-07
  hidden rotation_43 -1 0 0 4.2461773302480677e-07
  hidden position_43_0 -2.0278535653293795e-06
  hidden rotation_44 0 -1 0 2.0278723044639504e-06
  hidden position_44_0 -2.0390976990722484e-06
  hidden rotation_45 0 -1 0 2.0390103439209537e-06
  hidden position_45_0 -2.2844534279173295e-06
  hidden rotation_46 0 -1 0 2.2844991560861875e-06
  hidden position_46_0 1.2780967150258952e-06
  hidden rotation_47 1 0 0 1.2780297781462628e-06
  hidden translation_48 0.02000050842024159 -0.006999305960058599 -0.040757443143498565
  hidden rotation_48 5.236635787979135e-07 3.064118000790755e-08 0.9999999999998624 -3.14158528492407
  hidden rotation_52 0 -0.7071067811865476 0.7071067811865476 1.4609218810530206e-06
  hidden position_52_0 1.1462332343999874e-06
  hidden rotation_53 1 0 0 1.1461308314221903e-06
  hidden position_53_0 -1.987660796249165e-06
  hidden rotation_54 0 -1 0 1.987616062227935e-06
  hidden position_54_0 -2.1029418940745113e-06
  hidden rotation_55 0 -1 0 2.1029123499773668e-06
  hidden position_55_0 -2.1070605451793627e-06
  hidden rotation_56 0 -0.9999999999999999 0 2.107131680764479e-06
  hidden position_56_0 1.905986459023244e-06
  hidden rotation_57 1 0 0 1.9059511369848355e-06
  hidden translation_58 0.02000038066324772 0.00700050523746043 -0.04076022439799751
  hidden rotation_58 5.239120911489524e-07 3.1670865138843734e-08 0.9999999999998623 -3.141585280343662
  translation -0.6587053652039199 -2.051012929174993 0.3324212466811263
  rotation -0.005022810262253163 -0.019013611923188654 -0.9998066082691711 0.038724141732574636
  name "red_right_attacker"
  controller "red_right_attacker_controller"
}
DEF RED_DEFENDER Robot_nao {
  hidden position_0_0 -0.11386245691953982
  hidden position2_0_0 0.273316649335447
  hidden position_0_1 -0.7119382109397152
  hidden position2_0_1 -0.7388423480556597
  hidden position_0_2 0.9037298849116999
  hidden position2_0_2 0.28228546315410163
  hidden position_0_3 -5.439488799090475e-07
  hidden position_0_4 -2.6499209026124637e-06
  hidden linearVelocity_0 -3.2442012749172862e-06 2.2796527281325666e-07 4.230042277082987e-06
  hidden angularVelocity_0 -6.341536586175393e-06 -9.034555961894786e-06 -1.59156732547191e-06
  hidden rotation_11 0.052577333799238256 0.9225256332634432 -0.3823376517457399 0.2959490238231216
  hidden linearVelocity_11 -4.376427649375518e-06 9.979627987054662e-07 4.363729818170191e-06
  hidden angularVelocity_11 -6.3415442171273505e-06 -9.034533742199615e-06 -1.5915643839517925e-06
  hidden rotation_18 0.01748183370484817 0.9996420132501952 -0.020254156003459284 1.6377725200694167e-06
  hidden position_18_0 0.38832583059960013
  hidden linearVelocity_18 -3.921194899907674e-06 5.812368217045761e-07 4.916727491238818e-06
  hidden angularVelocity_18 -6.341504653501051e-06 -9.034496239712884e-06 -1.5915349085590213e-06
  hidden position_19_0 1.3688603560074624
  hidden linearVelocity_19 -3.944503421040121e-06 4.7033389251738317e-07 5.628840650326673e-06
  hidden angularVelocity_19 -3.604871674637624e-06 -9.243968686917644e-06 -1.5930757253228083e-06
  hidden rotation_20 0 0 1 6.063875540904964e-07
  hidden position_20_0 -0.06146325968933162
  hidden linearVelocity_20 -3.951377194881328e-06 4.663483320756237e-07 6.136996252362045e-06
  hidden angularVelocity_20 -3.6031678003874623e-06 -9.244017137608698e-06 1.4341245028200233e-06
  hidden rotation_21 1 0 0 2.0647654623614288e-07
  hidden position_21_0 0.16638276986731676
  hidden position_21_1 0.1663827698643709
  hidden position_21_2 0.16638276998575752
  hidden linearVelocity_21 -3.974280065160212e-06 5.646218628114383e-07 6.739435894706556e-06
  hidden angularVelocity_21 -2.562769280790588e-06 -9.323651311430218e-06 1.433537401872369e-06
  hidden rotation_22 0 1 0 0.79361723013268
  hidden position_22_0 0.1663827706846335
  hidden linearVelocity_22 -3.83720950120815e-06 5.871092550828371e-07 7.1347039461393e-06
  hidden angularVelocity_22 -2.5627692630254527e-06 -9.32365125217226e-06 1.4335373986680263e-06
  hidden rotation_23 0 1 0 0.7936172293153652
  hidden position_23_0 0.16638277111078587
  hidden linearVelocity_23 -3.727805004539609e-06 5.650057434032983e-07 7.187699526369059e-06
  hidden angularVelocity_23 -2.5627692828061543e-06 -9.323651298054571e-06 1.4335374003725667e-06
  hidden rotation_24 0 1 0 0.79361722888921
  hidden linearVelocity_24 -3.6148737688303447e-06 5.274793967585311e-07 7.1458964617801125e-06
  hidden angularVelocity_24 -2.562769281052629e-06 -9.323651304747915e-06 1.4335373988027893e-06
  hidden rotation_25 0 0.9999999999999999 0 0.7936172301356275
  hidden position_25_0 0.16638277068447946
  hidden linearVelocity_25 -3.874383748747835e-06 5.899598228030406e-07 7.086784998320687e-06
  hidden angularVelocity_25 -2.562769284790709e-06 -9.32365116297547e-06 1.4335374156801783e-06
  hidden rotation_26 0 0.9999999999999999 0 0.793617229315518
  hidden position_26_0 0.16638277111111302
  hidden linearVelocity_26 -3.7649792167888447e-06 5.678563080630584e-07 7.139780577128288e-06
  hidden angularVelocity_26 -2.5627692845958142e-06 -9.323651332377185e-06 1.4335374047334438e-06
  hidden rotation_27 0 0.9999999999999999 0 0.7936172288888831
  hidden linearVelocity_27 -3.6520479631232183e-06 5.303299606945977e-07 7.097977510788131e-06
  hidden angularVelocity_27 -2.5627692799914448e-06 -9.323651324530771e-06 1.4335374039436982e-06
  hidden rotation_28 0 1 0 0.16638276998575688
  hidden position_28_0 0.1663827692262218
  hidden linearVelocity_28 -3.691539437304393e-06 5.169107052258451e-07 6.938143928142261e-06
  hidden angularVelocity_28 -2.562769280542897e-06 -9.323651362187805e-06 1.4335374061945976e-06
  hidden rotation_29 0 -1 0 0.7936172307737781
  hidden linearVelocity_29 -3.7113656493069024e-06 5.408059347543202e-07 7.059026561048628e-06
  hidden angularVelocity_29 -2.562769273951412e-06 -9.32365113310052e-06 1.433537403637822e-06
  hidden rotation_30 -0.02529984902349859 0.9994812480647108 -0.01992868300706886 1.6377725200694167e-06
  hidden position_30_0 -0.9713518418682723
  hidden linearVelocity_30 -3.6303489950993296e-06 5.589988144672444e-07 3.884384730714023e-06
  hidden angularVelocity_30 -6.341544731150631e-06 -9.034525632067562e-06 -1.5915444716418837e-06
  hidden position_31_0 -0.8894455608800169
  hidden linearVelocity_31 -3.5860148289558268e-06 4.4215041274168734e-07 4.356309964755974e-06
  hidden angularVelocity_31 -9.07783487810504e-06 -8.824671740619833e-06 -1.590001351664847e-06
  hidden rotation_32 0 0 0.9999999999999999 6.027146701859462e-07
  hidden position_32_0 0.38699999609412394
  hidden linearVelocity_32 -3.6005581558247914e-06 4.4369695600071993e-07 4.832972722368708e-06
  hidden angularVelocity_32 -9.076132606838151e-06 -8.824732123574626e-06 1.4368850826128178e-06
  hidden rotation_33 -1 0 0 1.9992005623875177e-07
  hidden position_33_0 0.1319999998458632
  hidden position_33_1 0.13199999984482816
  hidden position_33_2 0.1320000003870614
  hidden linearVelocity_33 -3.6114335572680295e-06 5.602410540231623e-07 5.477078905340257e-06
  hidden angularVelocity_33 -1.0116459139837342e-05 -8.744945160574133e-06 1.4374723332898552e-06
  hidden rotation_34 0 1 0 0.828000000154134
  hidden position_34_0 0.13200000017886773
  hidden linearVelocity_34 -3.4827491696237234e-06 4.87905040882438e-07 5.959509113046248e-06
  hidden angularVelocity_34 -1.0116459139614621e-05 -8.744945200530623e-06 1.4374723347427937e-06
  hidden rotation_35 0 1 0 0.8279999998211327
  hidden position_35_0 0.13200000008963247
  hidden linearVelocity_35 -3.378674451443868e-06 3.748231330524656e-07 6.006707122718823e-06
  hidden angularVelocity_35 -1.011645912216969e-05 -8.744945112061772e-06 1.4374723309430539e-06
  hidden rotation_36 0 1 0 0.8279999999103619
  hidden linearVelocity_36 -3.2771652071301363e-06 2.4929775307163537e-07 5.95583174791607e-06
  hidden angularVelocity_36 -1.0116459122058457e-05 -8.744945026013982e-06 1.4374723333897475e-06
  hidden rotation_37 0 1 0 0.8280000001551673
  hidden position_37_0 0.13200000017927457
  hidden linearVelocity_37 -3.5200247936034265e-06 4.907728978424012e-07 5.714618839518168e-06
  hidden angularVelocity_37 -1.0116459117759577e-05 -8.744944995163289e-06 1.4374723242247172e-06
  hidden rotation_38 0 1 0 0.8279999998207198
  hidden position_38_0 0.13200000008986107
  hidden linearVelocity_38 -3.415950041433089e-06 3.7769098744852324e-07 5.7618169370209645e-06
  hidden angularVelocity_38 -1.0116459130926447e-05 -8.744944955272366e-06 1.437472329017017e-06
  hidden rotation_39 0 1 0 0.8279999999101323
  hidden linearVelocity_39 -3.3144407799584082e-06 2.521656060857449e-07 5.710941606446173e-06
  hidden angularVelocity_39 -1.0116459123605155e-05 -8.744945006286683e-06 1.4374723318820865e-06
  hidden rotation_40 0 1 0 0.1320000003870601
  hidden position_40_0 0.13200000006604082
  hidden linearVelocity_40 -3.3503996621703324e-06 2.876572002518615e-07 5.666146297296242e-06
  hidden angularVelocity_40 -1.0116459125142564e-05 -8.744945057306161e-06 1.4374723292899573e-06
  hidden rotation_41 0 -1 0 0.8279999999339532
  hidden linearVelocity_41 -3.374191292487807e-06 3.338803708280084e-07 5.784978163691022e-06
  hidden angularVelocity_41 -1.01164591215399e-05 -8.744945023726766e-06 1.4374723340096274e-06
  hidden rotation_42 0 -0.7071067811865475 -0.7071067811865475 1.636687543123527e-06
  hidden position_42_0 2.0024742792135297e-06
  hidden linearVelocity_42 -2.332768912936177e-06 -4.1581924606537744e-07 4.434692026150795e-06
  hidden angularVelocity_42 -6.255362541931571e-06 -7.923138576643824e-06 -4.889583870385559e-07
  hidden rotation_43 1 0 0 1.522837435131255e-06
  hidden position_43_0 -3.092507416619424e-06
  hidden linearVelocity_43 -2.047003596788172e-06 -5.670101202017125e-07 4.4407968634222155e-06
  hidden angularVelocity_43 1.8708187220489957e-06 -8.557112340285906e-06 -5.170773150110463e-07
  hidden rotation_44 0 -1 0 1.797304902471913e-06
  hidden position_44_0 -3.144803830423526e-06
  hidden linearVelocity_44 -1.5983323960592087e-06 -4.836529188153044e-07 4.588484983388252e-06
  hidden angularVelocity_44 1.870785831770318e-06 -8.562678366253735e-06 -5.406370791910605e-07
  hidden rotation_45 0 -1 0 1.8407608946183686e-06
  hidden position_45_0 -3.26501973204059e-06
  hidden linearVelocity_45 -7.450032479069482e-07 -3.0569926383763603e-07 4.632398924643777e-06
  hidden angularVelocity_45 1.870551800445967e-06 -8.566142974333303e-06 -5.641962561270312e-07
  hidden rotation_46 0 -1 0 1.9001171743030166e-06
  hidden position_46_0 1.5023558132961383e-07
  hidden linearVelocity_46 -3.09174719377113e-07 -2.154257815674168e-07 4.620675543613807e-06
  hidden angularVelocity_46 1.8701425969889949e-06 -8.567297969242065e-06 -5.87754483871496e-07
  hidden rotation_47 1 0 0 5.763494850937659e-07
  hidden linearVelocity_47 6.140165666663235e-08 -9.805517901315939e-08 4.835473603378659e-06
  hidden angularVelocity_47 3.666535981349687e-06 -8.708479459393122e-06 -6.123077270059453e-07
  hidden translation_48 0.02001004280410157 -0.006999711554639132 -0.040529185120478095
  hidden rotation_48 2.4289410279589366e-06 2.124433539394605e-08 0.99999999999705 -3.141585267625585
  hidden linearVelocity_48 1.6520170451646612e-07 -6.389389897301534e-08 4.791573650637971e-06
  hidden angularVelocity_48 3.66607271352317e-06 -8.710087126652047e-06 -6.362040020740149e-07
  hidden rotation_52 0 -0.7071067811865475 0.7071067811865475 2.563692708922217e-07
  hidden position_52_0 2.213986883812019e-07
  hidden linearVelocity_52 -2.289845656230993e-06 -4.0999042034948354e-07 4.014156548491971e-06
  hidden angularVelocity_52 -6.252694253723245e-06 -7.851642081406328e-06 -2.7494228979213396e-06
  hidden rotation_53 1 0 0 2.1544420208845964e-06
  hidden position_53_0 -3.1745323165109127e-06
  hidden linearVelocity_53 -2.050672816794916e-06 -5.415849687250718e-07 3.9141940299032564e-06
  hidden angularVelocity_53 3.4143667361866085e-06 -8.583446076930918e-06 -2.7354448277408457e-06
  hidden rotation_54 0 -0.9999999999999999 0 1.8390713477780885e-06
  hidden position_54_0 -3.165732348404137e-06
  hidden linearVelocity_54 -1.6810509163347145e-06 -4.2321032150520527e-07 4.08959168469257e-06
  hidden angularVelocity_54 3.413812660102512e-06 -8.575820772124145e-06 -2.716033685012043e-06
  hidden rotation_55 0 -0.9999999999999999 0 1.8647301448537378e-06
  hidden position_55_0 -3.2681303509106547e-06
  hidden linearVelocity_55 -9.101928751687261e-07 -1.0646822027744445e-07 4.149971746951039e-06
  hidden angularVelocity_55 3.4135048999181775e-06 -8.571764519502308e-06 -2.6966233358982753e-06
  hidden rotation_56 0 -1 0 1.912231825356354e-06
  hidden position_56_0 -9.651389308887631e-07
  hidden linearVelocity_56 -5.627841577781492e-07 6.289646363865458e-08 4.142313795126377e-06
  hidden angularVelocity_56 3.4135029117167395e-06 -8.570836616396748e-06 -2.6772145601988302e-06
  hidden rotation_57 -1 0 0 5.21327724591969e-07
  hidden linearVelocity_57 -2.8135266441943535e-07 2.9322702140676927e-08 4.386205331532814e-06
  hidden angularVelocity_57 3.399911927664534e-07 -8.335459527850156e-06 -2.656097249420795e-06
  hidden translation_58 0.0200100275301498 0.0070000142570465496 -0.040530966665886424
  hidden rotation_58 2.4225840986396674e-06 2.3142197589836137e-08 0.9999999999970653 -3.1415852496510412
  hidden linearVelocity_58 -2.5068731032726586e-07 4.424807846880246e-08 4.378984533973107e-06
  hidden angularVelocity_58 3.39797431270263e-07 -8.336698520092636e-06 -2.636891192736461e-06
  translation -0.5858019070538779 -0.11229851686005535 0.3322179821199428
  rotation 0.0009043951489088448 0.007295736064081905 -0.999972976787222 0.07647283380515245
  name "red_defender"
  controller "models"
  channel 5
}
DEF RED_GOALKEEPER Robot_nao {
  hidden position_0_0 -0.19735591184625395
  hidden position2_0_0 -0.06684139576074603
  hidden position_0_1 -0.14997346363512135
  hidden position2_0_1 -0.02245580127464017
  hidden position_0_2 0.8645468236174667
  hidden position2_0_2 0.2797508126300367
  hidden position_0_3 -1.0528048474630825e-08
  hidden position_0_4 1.3846335416763716e-07
  hidden rotation_11 -0.03165975065741681 -0.3197970916388068 -0.9469569580332945 0.2083329456026548
  hidden rotation_18 0.07151367680114362 -0.9864489887160139 0.1476624078482233 1.2147303812936079
  hidden position_18_0 -0.1758867886332977
  hidden rotation_19 -1 0 0 0.5718867314766019
  hidden position_19_0 1.3172877150120599
  hidden rotation_20 0 0 1 0.6962878107925325
  hidden position_20_0 -0.05982478859420689
  hidden rotation_21 -1 0 0 0.029824790224547384
  hidden position_21_0 0.16698992964450995
  hidden position_21_1 0.16698992964216466
  hidden position_21_2 0.16698992999931134
  hidden rotation_22 0 1 0 0.9590100707817265
  hidden position_22_0 0.16698992943300184
  hidden rotation_23 0 1 0 0.9590100699331042
  hidden position_23_0 0.16698992949783176
  hidden rotation_24 0 1 0 0.9590100702004279
  hidden rotation_25 0 1 0 0.9590100707840767
  hidden position_25_0 0.1669899294315576
  hidden rotation_26 0 1 0 0.9590100699347199
  hidden position_26_0 0.16698992949756464
  hidden rotation_27 0 1 0 0.9590100702006691
  hidden rotation_28 0 1 0 0.16698992999931234
  hidden position_28_0 0.16698992991207343
  hidden rotation_29 0 -1 0 0.9590100699149986
  hidden rotation_30 -0.20572430231950284 -0.9657393559055802 -0.15819294513418994 0.10504564909723903
  hidden position_30_0 -0.9406698969821188
  hidden rotation_31 -1 0 0 0.06166998318493941
  hidden position_31_0 -0.8680330019353227
  hidden rotation_32 0 0 1 0.12996684636639233
  hidden position_32_0 0.3869998994491127
  hidden rotation_33 -0.9999999999999999 0 0 9.884312124119402e-08
  hidden position_33_0 0.13199999984661756
  hidden position_33_1 0.1319999998462307
  hidden position_33_2 0.13200000023716488
  hidden rotation_34 0 1 0 0.8280000001533823
  hidden position_34_0 0.13200000009641627
  hidden rotation_35 0 1 0 0.8279999999035831
  hidden position_35_0 0.13200000005112603
  hidden rotation_36 0 1 0 0.8279999999488741
  hidden rotation_37 0 1 0 0.828000000153767
  hidden position_37_0 0.13200000006341098
  hidden rotation_38 0 1 0 0.8279999999365883
  hidden position_38_0 0.1320000000515363
  hidden rotation_39 0 1 0 0.8279999999484634
  hidden rotation_40 0 1 0 0.1320000002371654
  hidden position_40_0 0.13200000003550383
  hidden rotation_41 0 -1 0 0.8279999999644932
  hidden rotation_42 0 -0.7071067811865476 -0.7071067811865476 8.275177107402635e-07
  hidden position_42_0 -1.135760439762642e-06
  hidden rotation_43 -1 0 0 7.586432731960971e-07
  hidden position_43_0 -1.19298720563896e-06
  hidden rotation_44 0 -1 0 1.6317962195878268e-06
  hidden position_44_0 -1.1941686804813288e-06
  hidden rotation_45 0 -1 0 1.6606607098287859e-06
  hidden position_45_0 -1.2260710447891897e-06
  hidden rotation_46 0 -1 0 1.718485742013328e-06
  hidden position_46_0 2.7322421024799487e-08
  hidden rotation_47 -1 0 0 4.1509831777763796e-07
  hidden translation_48 0.020000290540178156 -0.007000212881647649 -0.040843466581099386
  hidden rotation_48 4.804783982052939e-07 -1.1654655588535128e-07 0.9999999999998779 -3.1415853430423266
  hidden rotation_52 0 0.7071067811865475 -0.7071067811865475 2.1696418193818693e-07
  hidden position_52_0 2.353495761879867e-06
  hidden rotation_53 1 0 0 2.0162322849730923e-06
  hidden position_53_0 -1.3101120970304692e-06
  hidden rotation_54 0 -1 0 1.7906211478472117e-06
  hidden position_54_0 -1.3864393711104903e-06
  hidden rotation_55 0 -1 0 1.8742320160306335e-06
  hidden position_55_0 -1.4356382811482762e-06
  hidden rotation_56 0 -1 0 1.9304912973287433e-06
  hidden position_56_0 1.1397282770121395e-06
  hidden rotation_57 1 0 0 5.930587274471662e-07
  hidden translation_58 0.02000011352095754 0.00699976052504625 -0.04083318579004333
  hidden rotation_58 4.91691560989153e-07 -1.2000156175406956e-07 0.999999999999872 -3.1415853466716626
  translation -4.401226137717444 -0.012576272911759565 0.33249247157802125
  rotation -0.004534561753705759 -0.0069654156517357495 0.9999654597707365 0.17474740597582367
  name "red_goalkeeper"
  controller "red_goalkeeper_controller"
  supervisor TRUE
}
RobocupSoccerBall {
  translation -0.10732532021313403 0 0.06967823204754754
  rotation 0 -1 0 1.533221108512396
}
Stadium_background {
  texture "stadium"
}
Statium_background_light {
  texture "stadium"
}
